Updated server script to check if server is existent

KeenanMcCarthy opened the pull request:
Updated the bin/server script to check if SERVER_OUTPUT_DIR exists before executing the run, debug, and checkpoint commands. If the server does not exist, the script issues a "no existing server" message and terminates with exit code 2.

fixes #18585

tjwatson commented on dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:
```diff
@@ -1475,6 +1475,11 @@ case "$ACTION" in
       exit $?
     fi
 
+    if [ ! -d "${SERVER_OUTPUT_DIR}" ]; then
```
Can we only do this `if [ ! -d "${SERVER_OUTPUT_DIR}" ] && [ ${ACTION} == "checkpoint" ]; then` instead?

That way we only change the case where a `checkpoint` is being done.

Commit:
Modified checkpoint command to avoid writing to log file if server does not exist

In the event of a checkpoint, the server script will check if the server.xml file exists in the SERVER_CONFIG_DIR. If the specified server is non-existant, then the console.log file will not be created. However, the server working directory will always be created whether or not the server is existant, this behavior is consistent with the "run" and "debug" actions.

tjwatson commented on dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:
```diff
@@ -1008,7 +1008,7 @@ javaCmd()
   elif [ $ACTION = "run" ] || [ $ACTION = "debug" ]; then
     exec "${JAVA_CMD}" "$@"
-  elif [ $ACTION = "checkpoint" ]; then
+  elif [ $ACTION = "checkpoint" ] && [ -d "${SERVER_OUTPUT_DIR}/server.xml" ]; then
```
`SERVER_CONFIG_DIR` should be used instead of `SERVER_OUTPUT_DIR` here.  The `SERVER_OUTPUT_DIR` defaults to the same spot as `SERVER_CONFIG_DIR` but it doesn't have to.

KeenanMcCarthy commented:
#build

tjwatson reviewed (approved):
@KeenanMcCarthy you can merge this one.

KeenanMcCarthy commented:
#build

tjwatson commented:
#libby
